src/DevTools/mockRelayExpectations.ts

```diff
@@ -0,0 +1,72 @@
+import { waitFor } from "@testing-library/react"
+import { OperationDescriptor } from "react-relay"
+import { MockEnvironment, MockPayloadGenerator } from "relay-test-utils"
+
+type ResolverHandler = {
+  /* Resolve operation with mock data in the mockResolvers format */
+  mockResolvers: MockPayloadGenerator.MockResolvers
+}
+type ErrorHandler = {
+  /* Reject operation with an error */
+  error: Error
+}
+type OperationConfig = {
+  /* The expected operation name */
+  name?: string
+  /* The operation variables */
+  variables?: any
+} & (ResolverHandler | ErrorHandler)
+
+/**
+ * Get the most recent pending relay operaion, optionally assert on its name or variables,
+ * and resolve or reject it with mock data or an error. A partially-applied version of this
+ * with the relay environment already included is also available as return
+ * value from `setupTestWrapperTL().renderWithRelay()`.
+ * @example
+ * ```tsx
+ * const mockRelayEnv = createMockEnvironment()
+ *
+ * await userEvent.click(screen.findByText("Submit"))
+ * await expectAndHandleOperationWithEnv(mockRelayEnv, {
+ *  name: "UpdateUserDefaultAddressMutation",
+ *  variables: { input: { addressID: "123" } },
+ *  mockResolvers: {
+ *   UpdateUserDefaultAddressMutation: () => ({ internalID: "123", isDefault: true }),
+ *  }
+ * )
+ *
+ * await expectAndHandleOperationWithEnv(mockRelayEnv, {
+ *   name: "grantAdminPrivilegesMutation",
+ *   error: new Error("You do not have permission to perform this action"),
+ * })
+ * ```
+ */
+export async function expectAndHandleOperationWithEnv(
```

Comment:
Can we name this something a little gentler? The name is saying what this is doing, but we're generally always "handling stuff with relay operations". What would be a friendly, terse way to express what this is doing? `resolveMutations` maybe? 

And we don't want assertions buried in lib code. Lib code _supports_ assertions, and we do what we can to make that as easy as possible. But when we start mixing the two together, it soon becomes brittle and hard to understand later. 

The main named export should be 1:1 with the file name.